oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java

```diff
@@ -79,6 +80,9 @@ public StorageEsInstaller(ModuleManager moduleManager, int indexShardsNumber, in
 
     @Override protected void createTable(Client client, Model tableDefine) throws StorageException {
         ElasticSearchClient esClient = (ElasticSearchClient)client;
+        if ((!tableDefine.isDeleteHistory() || tableDefine.getName().endsWith(Downsampling.Month.getName()) || tableDefine.getName().endsWith(Downsampling.Day.getName())) && esClient.getCreateByDayIndexes().contains(tableDefine.getName())) {
```

Comment:
This condition checks look like complex, especially when you based on table name, do you have better way to do that?